src/mlagility/common/filesystem.py

```diff
@@ -39,16 +40,22 @@ class CacheError(exc.Error):
 
 
 def _load_yaml(file) -> Dict:
-    if os.path.isfile(file):
-        with open(file, "r", encoding="utf8") as stream:
-            return yaml.load(stream, Loader=yaml.FullLoader)
-    else:
-        return {}
+    with _get_file_lock(file):
+        if os.path.isfile(file):
+            with open(file, "r", encoding="utf8") as stream:
+                return yaml.load(stream, Loader=yaml.FullLoader)
+        else:
+            return {}
 
 
 def _save_yaml(dict: Dict, file):
-    with open(file, "w", encoding="utf8") as outfile:
-        yaml.dump(dict, outfile)
+    with _get_file_lock(file):
```

Comment:
Is it possible to add a test to make sure the lock works as intended in https://github.com/groq/mlagility/blob/main/test/database.py?

Perhaps something like https://github.com/groq/mlagility/blob/319f0d00984da9bd8c3c233365cde8cb888f9dbb/test/database.py#LL59C19-L59C19, but instead of adding 3 builds it adds like 50 using 50 separate subprocesses?

Reply:
Added the test! Fun to see how terribly it fails without the added changes.